modules/platforms/dotnet/Apache.Ignite.Core/Impl/Common/FutureConverter.cs

```diff
@@ -35,28 +35,43 @@ internal class FutureConverter<T> : IFutureConverter<T>
         /** Converting function. */
         private readonly Func<BinaryReader, T> _func;
 
+        /** True if should register type both for dotnet and java platforms. */
+        private readonly bool _registerSameJavaType;
+
         /// <summary>
         /// Constructor.
         /// </summary>
         /// <param name="marsh">Marshaller.</param>
         /// <param name="keepBinary">Keep binary flag.</param>
         /// <param name="func">Converting function.</param>
+        /// <param name="registerSameJavaType">True if should register type both for dotnet and java platforms.</param>	
         public FutureConverter(Marshaller marsh, bool keepBinary,
-            Func<BinaryReader, T> func = null)
+            Func<BinaryReader, T> func = null, bool registerSameJavaType = false)
         {
             _marsh = marsh;
             _keepBinary = keepBinary;
             _func = func ?? (reader => reader == null ? default(T) : reader.ReadObject<T>());
+            _registerSameJavaType = registerSameJavaType;
         }
 
         /// <summary>
         /// Read and convert a value.
         /// </summary>
         public T Convert(IBinaryStream stream)
         {
-            var reader = stream == null ? null : _marsh.StartUnmarshal(stream, _keepBinary);
+            var locRegisterSameJavaType = Marshaller.RegisterSameJavaType.Value;
+            Marshaller.RegisterSameJavaType.Value = _registerSameJavaType;
+
+            try
+            {
+                var reader = stream == null ? null : _marsh.StartUnmarshal(stream, _keepBinary);
```

Comment:
I think all changes in this file are unnecessary, since we don't write anything, only read.

Reply:
These changes required for reading binary object of unregistered type and dynamically resolve those type.
Please, take a look at `ComputeApiTypeAutoRegisterTest#TestEchoTasksAutoRegisterType`